sympy/physics/mechanics/abstract_body.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+
+from sympy.core.backend import Symbol, sympify
+from sympy.physics.vector import Point
+
+__all__ = ['_Body']
+
+
+class _Body(ABC):
+    """Abstract class for body type objects."""
+    def __init__(self, name, masscenter=None, mass=None):
+        if not isinstance(name, str):
+            raise TypeError('Supply a valid name.')
```

Comment:
Do we want to do any further error checking here? For example ensuring that the name has nonzero length? Or does this break backwards compatibility?

Reply:
Okay, we can leave as is for now. But perhaps when we implement `System` it'll be a good time to do error checking on name clashes.

Reply:
Yeah, in `System` there will definitely be checking of all these kind of things.